posts/2024/2024_06_28_Inigo_week_5.rst

```diff
@@ -0,0 +1,31 @@
+Week 5 into GSoC 2024: Vacation, starting with the conditional AutoEncoder
+==========================================================================
+
+
+.. post:: June 28 2024
+    :author: Iñigo Tellaetxe
+    :tags: google
+    :category: gsoc
+
+
+
+What I did this week
+~~~~~~~~~~~~~~~~~~~~
+
+Hi everyone! This week I have been on vacation, so I have not been able to work on the project as much as the previous weeks. However, I have been thinking about the next steps to take and I have decided to start with the conditional AutoEncoder. I have been reading some papers and I have found some interesting ideas that would be nice to implement.
+
+While stuck in the Munich airport, I started to write some code for this (the weather was not very good and my flight was delayed, so I lost my connecting flight and I had to sleep in the airport). I found an implementation of a regression variational AutoEncoder `in this paper <https://doi.org/10.1007/978-3-030-32245-8_91>`_, where the authors implement a way to manipulate the latent space so that the samples (streamlines) that get projected into it are organized along a desired scalar parameter.
+I thought this could be a good starting point for my conditional AutoEncoder because it basically provides a way to sample from the latent space in a controlled manner, where you can select the age of the streamlines you want to generate. Also, having the variational part makes the latent space more regularized, so our model is more resilient against overfitting, which is the main flaw of ordinary AutoEncoders.
```

Comment:
Also, I am all that comfortable saying that overfitting is an inherent property to AEs. Do you have a scientific reference that supports this?

Reply:
Thnks for pointing this out. To be precise, I should rephrase this to **_"...overfitting, what could happen when non-regularized (vanilla) AutoEncoders learn the identity function, which leads to lack of generalizability outside the test set."_**

This is explained in [this NeurIPS 2020 paper](https://proceedings.neurips.cc/paper/2020/hash/e33d974aae13e4d877477d51d8bafdc4-Abstract.html) ([PDF](https://proceedings.neurips.cc/paper_files/paper/2020/file/e33d974aae13e4d877477d51d8bafdc4-Paper.pdf))
![image](https://github.com/dipy/dipy.org/assets/92918634/7476324e-21d2-4396-8702-dbcd20f483ce)


In conclusion, yes, saying that _"AEs main flaw is (natural) overfitting"_ is too much. However, saying that **_they might overfit towards the identity function_**, which is the effect I was aware of at the time I wrote the blog, is more rigurous.

What do you think @jhlegarreta ?

Reply:
Thanks for the reference.

"which leads to lack of generalizability outside the test set." outside the training set ?

Reply:
True, sorry. Yes, I meant "outside the **training** set"